docs/advanced/api_examples.yml

```diff
@@ -143,3 +143,18 @@
     HTTP/1.1 201 Created
 
     {"dataset_id": "886ed03a-5606-453a-94a9-a1cbaf35164c", "primary": "%(NODE_0)s", "metadata": {"name": "demo", "owner": "alice"}}
+
+-
+  id:
+    "get state datasets"
+
+  doc: |
+    Get a list of all datasets in a deployment.
+
+  request: |
+    GET /v1/state/datasets HTTP/1.1
+
+  response: |
+    HTTP/1.1 200 OK
+
+    [{"dataset_id": "47440eff-e933-4de0-b56c-d3469b61421f", "primary": "%(NODE_0)s", "maximum_size": 1073741824, "metadata": {}}]
```

Comment:
If you give this example a requirement of one of the examples that creates a dataset then ... maybe it might validate if we ever get to building a validator for the examples.  Though, more likely, it would also need to wait for the dataset to actually be created.  That could be another requirement that waits for an operation to complete and the API for that doesn't exist yet so you can't write the example...

So I guess nevermind. :smile:  But maybe later that'll be a strategy for making this example validate-able.